lib/clustering/random_clustering.cpp

```diff
@@ -0,0 +1,35 @@
+/// \file
+/// Maintainer: Luzian Hug
+/// Created: 25.03.2018
+///
+///
+
+
+#include "random_clustering.h"
+#include <stdlib.h>
+namespace MouseTrack {
+
+RandomClustering::RandomClustering(int k) : _k(k) {
+    //empty
+}
+
+std::vector<Cluster> RandomClustering::operator()(const PointCloud& cloud) const {
+
+	std::vector<Cluster> clustering = std::vector<Cluster>(_k);
+
+	//Assign random cluster to each point in cloud
+	for(PointIndex i=0; i<cloud.size(); i++) {
+		//Assign index i to random cluster in clustering
+		int cluster = rand() % _k;
```

Comment:
You might want to avoid the usage of `rand()` as much as possible without a seed value. Assume there are cases where `rand` can produce invalid input. Since it is random, this only happens sometimes and you have no clue, what pattern causes the problem. To use a seed, see [srand](http://www.cplusplus.com/reference/cstdlib/srand/).

But since this is a mock, it's ok.